### PR TITLE
add m68k arch support

### PR DESCRIPTION
**Pull request by knickish**
The M68k values were already included, this PR adds the `enum Architecture` entry for `M68k` and fills out the relevant match statements

### CODE REVIEW

File: src/write/elf/object.rs [hunk 1]
```diff
@@ -260,6 +261,16 @@ impl<'a> Object<'a> {
...
             },
+            Architecture::M68k => match (kind, encoding, size) {
+                (RelocationKind::Absolute, _, 8) => elf::R_68K_8,
+                (RelocationKind::Absolute, _, 16) => elf::R_68K_16,
+                (RelocationKind::Absolute, _, 32) => elf::R_68K_32,
+                (RelocationKind::Relative, _, 16) => elf::R_68K_PC16,
+                (RelocationKind::Relative, _, 32) => elf::R_68K_PC32,
+                _ => {
+                    return Err(Error(format!("unimplemented relocation {:?}", reloc)));
+                }
```
Comment (philipc) [inline src/write/elf/object.rs:272]: Please use the same code style as the other match arms.

**Review by philipc — CHANGES_REQUESTED** [review]
Please also update https://github.com/gimli-rs/object/blob/61dc7b00e0f8ce44d31e5555b124256a50fdeafd/src/read/elf/relocation.rs#L244 and https://github.com/gimli-rs/object/blob/61dc7b00e0f8ce44d31e5555b124256a50fdeafd/tests/round_trip/mod.rs#L260

**Comment by knickish** [comment]
Thanks, had missed those functions. I believe I've gotten everything and matched the formatting.

**Review by philipc — APPROVED** [review]
Thanks!

**Pull request merged**